### DISCUSSION TITLE
Iconify installed a lot of system apps

### DISCUSSION THREAD

**Discussion by NikunjK8** [Mahmud0808/Iconify#492] [Q&A]
I'm not using Iconify anymore, how can i remove them? Does Iconify itself has any setting? They're installed as system apps
![Screenshot_2024-04-26-13-40-49-290_com mixplorer](https://github.com/Mahmud0808/Iconify/assets/99175083/d5499d54-2f7e-4b30-8aa4-102fdc879a03)

**Comment by Mahmud0808** [comment]
Remove module from magisk/apatch/kernelsu.

**Reply by e-t-l** [reply]
Is there a reason Iconify has to install ALL the system overlay apps it generates? Why not just install whichever overlay app is currently enabled?

**Reply by Mahmud0808** [reply]
That means you'll need to reboot each time you enable something. Still, Iconify now generates around 50 overlays instead of the 300+ it did before v8, which I see as a huge improvement.

**Reply by e-t-l** [reply]
That's good to know! Is there a way to view which overlap apps are currently enabled?

**Reply by Mahmud0808** [reply]
Run in termux with root privilege: `cmd overlay list | grep '\[x\] IconifyComponent'`

**Reply by e-t-l** [reply]
Awesome thank you! I'm asking because I am working on modifying a custom ROM to be rootless, and I'd like to include some basic Iconify overlays (e.g. for Notifications or Immersive Navbar) as defaults in the ROM. 

I plan to sign these Iconify overlay apks with the system key and include them in the `/product/overlays` directory in the ROM image. But, I'm not sure if I would need to do something further to make them "active." Do you know how I would activate the overlays from here (without the whole Iconify app installed, and without using a root shell)? Would it work better to install Iconify \*as a system app\* in the image?

**Reply by Mahmud0808** [reply]
> Awesome thank you! I'm asking because I am working on modifying a custom ROM to be rootless, and I'd like to include some basic Iconify overlays (e.g. for Notifications or Immersive Navbar) as defaults in the ROM.
> 
> I plan to sign these Iconify overlay apks with the system key and include them in the `/product/overlays` directory in the ROM image. But, I'm not sure if I would need to do something further to make them "active." Do you know how I would activate the overlays from here (without the whole Iconify app installed, and without using a root shell)? Would it work better to install Iconify *as a system app* in the image?

You can use static overlays if you want them to always be enabled, or else make a UI in settings to trigger Overlay Manager Service.

**Comment by NikunjK8** [comment]
Thanks
